src/states/node.rs

```diff
@@ -2021,6 +2114,27 @@ impl Node {
         self.merge_if_necessary();
     }
 
+    pub fn section_list_signatures(&self,
+                                   prefix: Prefix<XorName>)
+                                   -> BTreeMap<PublicId, sign::Signature> {
+        if let Some(&(ref section_list, ref signatures)) =
+            self.section_list_sigs.get_signatures(prefix) {
+            let data = if let Ok(data) = serialisation::serialise(section_list) {
+                data
+            } else {
+                return Default::default();
+            };
+            signatures.iter()
+                .filter(|&(pub_id, sig)| {
+                    sign::verify_detached(sig, &data, pub_id.signing_public_key())
```

Comment:
Wouldn't it be more efficient to verify the signatures when they come in, and never put invalid ones into the `section_list_sigs` in the first place?